sdv/multi_table/hma.py

```diff
@@ -277,7 +270,10 @@ def _extract_parameters(self, parent_row, table_name, foreign_key):
         num_rows_key = f'{prefix}num_rows'
         if num_rows_key in flat_parameters:
             num_rows = flat_parameters[num_rows_key]
-            flat_parameters[num_rows_key] = min(self._max_child_rows[num_rows_key], num_rows)
+            flat_parameters[num_rows_key] = min(
+                self._max_child_rows[num_rows_key],
+                math.ceil(num_rows)
```

Comment:
why do we need to do this? `num_rows` should be an int right?

Reply:
Yes, but in some scenarios which has to generate the primary key it goes `0.3` for example, and that rounded goes to `0` which then ends up failing from `rdt` -> https://github.com/sdv-dev/RDT/blob/e4fb0e5960feb8e829c8d8f5f64abdade8f8eadf/rdt/hyper_transformer.py#L712